bst.py

```diff
@@ -0,0 +1,179 @@
+            return "digraph G{\n%s}" % ("" if self.val is None else (
+                "\t%s;\n%s\n" % (
+                    self.val,
+                    "\n".join(self._get_dot())
+                )
+            ))
+
+        def _get_dot(self):
+            """recursively prepare a dot graph entry for this node."""
+            if self.left is not None:
+                yield "\t%s -> %s;" % (self.val, self.left.val)
+                for i in self.left._get_dot():
+                    yield i
+            elif self.right is not None:
+                r = random.randint(0, 1e9)
+                yield "\tnull%s [shape=point];" % r
+                yield "\t%s -> null%s;" % (self.val, r)
+            if self.right is not None:
+                yield "\t%s -> %s;" % (self.val, self.right.val)
+                for i in self.right._get_dot():
+                    yield i
+            elif self.left is not None:
+                r = random.randint(0, 1e9)
+                yield "\tnull%s [shape=point];" % r
+                yield "\t%s -> null%s;" % (self.val, r)
+
+
+class BST(object):
+    def __init__(self, val=None):
+        self._size = 0
+        self.root = None
+        if val is not None:
+            self.insert(val)
+        else:
+            self.root = None
+
+    def insert(self, val):
+        """Insert val into BST if not present. Always returns None."""
+        if self.root is None:
+            self.root = Node(val)
+            self._size += 1
+        else:
+            new_parent = self._find(val)
+            if new_parent.val > val:
+                new_parent.left = Node(val, new_parent)
+                self._size += 1
+            elif new_parent.val < val:
+                new_parent.right = Node(val, new_parent)
+                self._size += 1
+            else:
+                return
+
+    def contains(self, val):
+        """Return True if val in BST. Else, return False."""
+        checkdis = self._find(val)
+        try:
+            if checkdis.val == val:
+                return True
+            else:
+                return False
+        except AttributeError:
+            return False
+
+    def size(self):
+        """Return int size of tree. Will return 0 if tree is empty."""
+        return self._size
+
+    def depth(self):
+        """Return the int number of levels in the tree.
+        Return 0 if tree is empty."""
+        if self.root is not None:
+            l, r = self._lr_levels()
+            return max([l, r]) + 1
+        else:
+            return 0
+
+    def balance(self):
+        """Return the difference between the number of nodes on the left with
+        respect to the right side.
+        --Negative if more on right
+        --Positive if more on left
+        --Zero if tree is perfectly balanced"""
+        if self.root is not None:
+            l, r = self._lr_levels()
+            return l - r
+        else:
+            return 0
```

Comment:
It's not really needed to have an `else` here.  If the condition for this `if` block is True, then the function will return.